plugin/storage/badger/options.go

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2018 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package badger
+
+import (
+	"flag"
+	"time"
+
+	"github.com/spf13/viper"
+)
+
+// Options store storage plugin related configs
+type Options struct {
+	primary *NamespaceConfig
+	others  map[string]*NamespaceConfig
+}
+
+// NamespaceConfig is badger's internal configuration data
+type NamespaceConfig struct {
+	namespace      string
+	SpanStoreTTL   time.Duration
+	ValueDirectory string
+	KeyDirectory   string
+	Ephemeral      bool // Setting this to true will ignore ValueDirectory and KeyDirectory
+	SyncWrites     bool
+}
+
+const (
+	suffixKeyDirectory   = ".directory-key"
+	suffixValueDirectory = ".directory-value"
+	suffixEphemeral      = ".ephemeral"
+	suffixSpanstoreTTL   = ".span-store-ttl"
+	suffixSyncWrite      = ".consistency"
+)
+
+// NewOptions creates a new Options struct.
+func NewOptions(primaryNamespace string, otherNamespaces ...string) *Options {
+	options := &Options{
+		primary: &NamespaceConfig{
+			namespace:      primaryNamespace,
+			SpanStoreTTL:   time.Hour * 72, // Default is 3 days
+			SyncWrites:     false,          // Performance over consistency
+			Ephemeral:      true,           // Default is ephemeral storage
+			ValueDirectory: "",
+			KeyDirectory:   "",
+		},
+		others: make(map[string]*NamespaceConfig, len(otherNamespaces)),
+	}
+
+	for _, namespace := range otherNamespaces {
+		options.others[namespace] = &NamespaceConfig{namespace: namespace}
+	}
+
+	return options
+}
+
+// AddFlags adds flags for Options
+func (opt *Options) AddFlags(flagSet *flag.FlagSet) {
+	addFlags(flagSet, opt.primary)
+	for _, cfg := range opt.others {
+		addFlags(flagSet, cfg)
+	}
+}
+
+func addFlags(flagSet *flag.FlagSet, nsConfig *NamespaceConfig) {
+	flagSet.Bool(
+		nsConfig.namespace+suffixEphemeral,
+		nsConfig.Ephemeral,
+		"Mark this storage ephemeral, data is stored in tmpfs (in-memory). Default is true.",
```

Comment:
The "default" part is added automatically by Viper. This is how it looks like when you run `SPAN_STORAGE_TYPE=badger go run ./cmd/collector/main.go --help`

```
--badger.ephemeral                  Mark this storage ephemeral, data is stored in tmpfs (in-memory). Default is true. (default true)
```